runtime/v1/shim/client/client.go

```diff
@@ -227,8 +227,7 @@ func connect(address string, d func(string, time.Duration) (net.Conn, error)) (n
 }
 
 func annonDialer(address string, timeout time.Duration) (net.Conn, error) {
-	address = strings.TrimPrefix(address, "unix://")
-	return net.DialTimeout("unix", "\x00"+address, timeout)
+	return dialer.Dialer("\x00"+address, timeout)
```

Comment:
didn't walk back to figure this out, but is there a reason the `TrimPrefix` of the `unix://` portion is no longer needed? Was it never needed?

Reply:
`dialer.Dialer` calls `pkg/dialer.dialer(string, time.Duration) (net.Conn, error)` which for unix always does the stripping

To be fair, I'm really not sure why the stripping was needed when dialing `\0...` in the first place.

Reply:
Let's keep `TrimPrefix` here. I don't think we should accept "\x00unix:///path/to/socket" .